### ISSUE TITLE
misc: remove any garbage and write test + github ci

### ISSUE THREAD

**Issue by andrewfurey21** [andrewfurey21/libtensor#15]

**Comment by andrewfurey21** [comment]
split tests in multiple files

**Comment by andrewfurey21** [comment]
remove separate stages when computing gradients depending on requires_grad, and check at the end.

**Comment by andrewfurey21** [comment]
remove any todos and add to issues
